lib/r10k/api/modules.rb

```diff
@@ -123,12 +125,22 @@ def write_module(module_name, env_map, path, opts={})
             git.clean(path, git_dir: git_dir, force: true)
           end
         when :forge
-          tarball_cachedir = cachedir_for_forge_module(mod[:source], opts[:cachedir])
+          release_slug = release_slug_from_module_slug_version(mod[:source], mod[:resolved_version])
+          cachedir = opts[:cachedir]
 
-          # TODO: cache tarball
+          unpack_tmpdir = cachedir_for_forge_module([release_slug, 'unpack'].join('-'), cachedir)
+
+          release_tarball = get_cached_forge_release(release_slug, cachedir, false, opts[:forge])
+
+          ## Unpack the files to a given directory
+          # @raise RuntimeError if it fails to extract the contents of the release tarball
+          begin
+            Dir.mktmpdir(release_slug, unpack_tmpdir) do |tmpdir|
```

Comment:
Currently, it goes to something like
/tmp/forge/gozar-amodule-0.1.0-unpack/gozar-amodule-0.1.020160217-31443-1ctpusx
Where "/tmp" is the specified cachedir.

I think `Dir.mktmpdir('', unpack_tmdir)` would be just as clear. Then it would be
/tmp/forge/gozar-amodule-0.1.0-unpack/20160217-31443-vr94g9